src/globus_cli/commands/flows/start.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import JSONStringOrFile, command, flow_id_arg
+from globus_cli.termio import Field, TextMode, display, formatters
+
+ROLE_TYPES = ("flow_viewer", "flow_starter", "flow_administrator", "flow_owner")
+
+
+@command("start", short_help="Start a flow")
+@flow_id_arg
+@click.option(
+    "--input",
+    "input_document",
+    type=JSONStringOrFile(),
+    help="""
+        The JSON input parameters used to start the flow.
+
+        The input document may be specified inline,
+        or it may be a path to a JSON file, prefixed with "file:".
+
+        Example: Inline JSON:
+
+        \b
+            --input '{"src": "~/source"}'
+
+        Example: Path to JSON file:
+
+        \b
+            --input file:parameters.json
+
+        If unspecified, the default is an empty JSON object ('{}').
+    """,
+)
+@click.option(
+    "--label",
+    type=str,
+    help="A label to give the run.",
+)
+@click.option(
+    "--manager",
+    "managers",
+    type=str,
+    multiple=True,
+    help="""
+        A principal that may manage the execution of the run.
+
+        This option can be specified multiple times
+        to create a list of run managers.
+    """,
+)
+@click.option(
+    "--monitor",
+    "monitors",
+    type=str,
+    multiple=True,
+    help="""
+        A principal that may monitor the execution of the run.
+
+        This option can be specified multiple time
+        to create a list of run monitors.
+    """,
+)
+@click.option(
+    "--tag",
+    "tags",
+    type=str,
+    multiple=True,
+    help="""
+        A tag to associate with the run.
+
+        This option can be used multiple times
+        to create a list of tags.
+    """,
+)
+@LoginManager.requires_login(LoginManager.FLOWS_RS)
+def start_command(
+    login_manager: LoginManager,
+    flow_id: uuid.UUID,
+    input_document: dict | None | t.Any,
+    label: str,
+    managers: tuple[str],
+    monitors: tuple[str],
+    tags: tuple[str],
+):
+    """
+    Start a flow.
+
+    FLOW_ID must be a UUID.
+    """
+
+    if input_document is None:
+        input_document = {}
+
+    flow_client = login_manager.get_specific_flow_client(flow_id)
+    response = flow_client.run_flow(
+        body=input_document,
+        label=label,
+        tags=list(tags),
+        run_managers=list(managers),
+        run_monitors=list(monitors),
+    )
+
+    auth_client = login_manager.get_auth_client()
+    principal_formatter = formatters.auth.PrincipalURNFormatter(auth_client)
```

Comment:
There is an optional performance improvement we can make here, but one which I'd like us to include.

What's needed is to iterate over all the values from `[response["run_owner"]] + response["run_managers"] + response["run_monitors"]` and pass them to `principal_formatter.add_item`.

To explain:

Each PrincipalFormatter gets a `globus_sdk.IdentityMap` which it contains. [PrincipalFormatter.resolved_ids](https://github.com/globus/globus-cli/blob/6fa67a6584ae181bf8f6801f6e587cf467c2b05a/src/globus_cli/termio/formatters/auth.py#L32).

[`add_item`](https://github.com/globus/globus-cli/blob/6fa67a6584ae181bf8f6801f6e587cf467c2b05a/src/globus_cli/termio/formatters/auth.py#L54) will attempt to parse those values and add them to that `IdentityMap`.

When `get()` or [`__getitem__`](https://github.com/globus/globus-sdk-python/blob/07d9b2cba29b7f8ef67ec27c24616c8232a7e382/src/globus_sdk/services/auth/identity_map.py#L244) is used on an `IdentityMap`, it attempts to lookup all of the identities which have been added in a single batch. (A single call to Auth.)

So by prepopulating the map, we get a single callout to Auth, rather than a separate call per element in each of these lists (with caching and cache hits).

Reply:
Thanks for explaining this! I've followed this pattern based on what I see in the nearby `show.py`, now.